src/solvers/eigen_solver.C

```diff
@@ -53,19 +53,74 @@ EigenSolver<T>::build(const Parallel::Communicator & comm,
   return UniquePtr<EigenSolver<T> >();
 }
 
-
 template <typename T>
 void EigenSolver<T>::set_solver_configuration(SolverConfiguration & solver_configuration)
 {
   _solver_configuration = &solver_configuration;
 }
 
+template <typename T>
+void EigenSolver<T>::set_position_of_spectrum (Real pos)
+{
+   if (pos>=0)
+     {
+        _position_of_spectrum=TARGET_MAGNITUDE;
+        _target_val=pos;
+     }
+  else
+    {
+       _position_of_spectrum= TARGET_REAL;
+        _target_val=pos;
+    }
+}
+
+template <typename T>
+void EigenSolver<T>::set_position_of_spectrum (Real pos, std::string ImagReal)
+{
+   if( ImagReal == "i")
+     {
+       _position_of_spectrum=TARGET_IMAGINARY;
+       _target_val=pos;
+     }
+   else if( ImagReal == "r")
+     {
+       _position_of_spectrum=TARGET_REAL;
+       _target_val=pos;
+     }
+   else if( ImagReal == "r")
```

Comment:
This `else if` block will never be hit because the condition is identical to the one before it...